src/main/java/edu/hm/hafner/analysis/parser/GccParser.java

```diff
@@ -36,6 +36,12 @@ public GccParser() {
         super(GCC_WARNING_PATTERN);
     }
 
+    @Override
+    protected boolean isLineInteresting(final String line) {
+        // Skip "In file included from" lines - these are include traces, not actual warnings
+        return !line.contains("In file included from") && super.isLineInteresting(line);
+    }
+
```

Comment:
When I remove this line, then the test still is green. 

You need to rethink your process, TDD is your friend: first, create a test case that fails. Then write the code until the tests are green. So your change actually does nothing.